src/tss2-esys/esys_iutil.c

```diff
@@ -611,13 +604,16 @@ iesys_restore_session_flags(ESYS_CONTEXT *esys_context)
 {
     LOG_DEBUG("Restoring session attribs");
 
-    if (esys_context->enc_session) {
-        LOG_DEBUG("Orig Session Attrs 0x%x, altered Attrs x%x",
-                  esys_context->enc_session->origSessionAttributes,
-                  esys_context->enc_session->sessionAttributes);
+    for (int i = 0; i < 3; i++) {
+        RSRC_NODE_T *session = esys_context->session_tab[i];
+        IESYS_SESSION *rsrc_session = &session->rsrc.misc.rsrc_session;
+        if (session == NULL)
```

Comment:
Looks like we first dereference the session pointer and then check if it is NULL afterwards?

Reply:
Totally true of course...
Monday then, or you just rewrite my patch...

Reply:
Monday is fine

Reply:
Fixed